Sources/Swift/Integrations/SessionReplay/SentryTouchTracker.swift

```diff
@@ -45,28 +46,29 @@ class SentryTouchTracker: NSObject {
     
     func trackTouchFrom(event: UIEvent) {
         guard let touches = event.allTouches else { return }
-        for touch in touches {
-            guard touch.phase == .began || touch.phase == .ended || touch.phase == .moved || touch.phase == .cancelled else { continue }
-            let info = trackedTouches[touch] ?? TouchInfo(id: touchId++)
-            let position = touch.location(in: nil).applying(scale)
-            let newEvent = TouchEvent(x: position.x, y: position.y, timestamp: event.timestamp, phase: touch.phase.toRRWebTouchPhase())
-            
-            switch touch.phase {
-            case .began:
-                info.startEvent = newEvent
-            case .ended, .cancelled:
-                info.endEvent = newEvent
-            case .moved:
-                // If the distance between two points is smaller than 10 points, we don't record the second movement.
-                // iOS event polling is fast and will capture any movement; we don't need this granularity for replay.
-                if let last = info.moveEvents.last, touchesDelta(last.point, position) < 10 { continue }
-                info.moveEvents.append(newEvent)
-                debounceEvents(in: info)
-            default:
-                continue
+        trackedTouchesLock.synchronized {
```

Comment:
`h`: This gets called from the main thread and is a huge risk for slowing it down. We already accept race conditions in the frames tracker when getting frames statistics from a background thread because, anyway, the statistics won't be 100% accurate, and it's better to have 99% accuracy over blocking the main thread. Instead, we should ensure we don't crash when reading tracked touches from a background thread and adding touches on the main thread.

Reply:
But this race condition can lead to crashes. By removing this synchronization, the test just not fail, it crashes.

This is not a matter of "we accept the consequences of race conditions". This is actually necessary.

Reply:
Also, the other two methods that also synchronize in this object are super fast locks. I don’t think we need to worry about this.

Reply:
OK, so `flushFinishedEvents` and `replayEvents` read `trackedTouches`, which get called on a bg thread. Therefore it doesn't matter if we also do this on a bg thread. We only need to ensure updating the values runs before reading them. So, we can move this logic here in the lock to a bg thread. Ideally, it uses the same sync queue as the code calling `flushFinishedEvents` and `replayEvents`, and then you wouldn't even need locks.

Reply:
I would prefer not blocking the main thread for `sendEvent:`. That gets called frequently when you for example scroll on a page.

Reply:
Yeah, we can do that. But Im keeping the lock and not managing dispatch queues in here

Reply:
Okay, I’m not keeping the lock, but I need to synchronize the access to `trackedTouches` for `flushFinishedEvents` and `replayEvents`. Otherwise, the test fails because the responsibility of calling these functions in the same dispatcher as `trackedTouches` would fall on the caller, which poses a significant future risk.